### PR TITLE
Eliminate mocking in net device

### PR DESCRIPTION
**Pull request by roypat**
## Changes

Removes mocking in net device code. Mocking in rust should not be done using `cfg(not(test))`, but rather using traits. However, in this specific case, there is no need for mocking, because we already have a separate framework for simulating tap traffic by writing to a socket that is connected to an actual tap device. So just remove mocking and switch the 3 tests that relied on it to this other framework.

## Reason

Removes the last major source of `cfg(not(test))` in our code base, and will make @bchalios's job easier (or so I've been told)

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this
  PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: src/vmm/src/devices/virtio/net/device.rs [hunk 1]
```diff
@@ -2065,19 +2066,18 @@ pub mod tests {
 
             // wait for 100ms to give the rate-limiter timer a chance to replenish
             // wait for an extra 100ms to make sure the timerfd event makes its way from the kernel
-            thread::sleep(Duration::from_millis(200));
+            thread::sleep(Duration::from_millis(1000));
```
Comment (ShadowCurse) [inline src/vmm/src/devices/virtio/net/device.rs:2068]: Need to adjust comment

File: src/vmm/src/devices/virtio/net/device.rs [hunk 2]
```diff
@@ -1948,11 +1948,10 @@ pub mod tests {
...
```
Comment (ShadowCurse) [inline src/vmm/src/devices/virtio/net/device.rs:1949]: Need to adjust comment

**Review by ShadowCurse — COMMENTED** [review]
LGTM

File: src/vmm/src/devices/virtio/net/device.rs (hunk 2 above)
Comment (roypat) [reply src/vmm/src/devices/virtio/net/device.rs:1949]: done

File: src/vmm/src/devices/virtio/net/device.rs (hunk 1 above)
Comment (roypat) [reply src/vmm/src/devices/virtio/net/device.rs:2068]: done

**Review by ShadowCurse — APPROVED** [review]

**Review by kalyazin — APPROVED** [review]

**Pull request merged**